packages/x-data-grid-pro/src/hooks/features/dataSource/useGridDataSource.ts

```diff
@@ -52,28 +69,53 @@ export const useGridDataSource = (
     | 'sortingMode'
     | 'filterMode'
     | 'paginationMode'
+    | 'pageSizeOptions'
     | 'treeData'
+    | 'lazyLoading'
   >,
 ) => {
+  const setStrategyAvailability = React.useCallback(() => {
+    apiRef.current.setStrategyAvailability(
+      'dataSource',
+      DataSourceRowsUpdateStrategy.Default,
+      props.unstable_dataSource && !props.lazyLoading ? () => true : () => false,
+    );
+  }, [apiRef, props.lazyLoading, props.unstable_dataSource]);
+
+  const [defaultRowsUpdateStrategyActive, setDefaultRowsUpdateStrategyActive] =
```

Comment:
since it is an already existing pattern, I don't mind keeping it like this.
if it starts creating problems in the future (with lazy loading of child nodes), we can refactor this part